lib/Support/ProgramRunner.cpp

```diff
@@ -75,6 +75,7 @@ int ProgramRunner::run(llvm::StringRef ProgramName,
   Log << DoLog;
 
   int ExitCode = llvm::sys::ExecuteAndWait(StringRefs[0], StringRefs);
+  Log << "Program exited with code " << ExitCode << "\n";
```

Comment:
This is illegal!

Either use `revng_log(Log, ...)` or...

```c
if (Log.isEnabled()) {
  Log << ... << DoLog;
}
```
If you don't use `revng_log` you have to finalize with `DoLog`. If you don't, the stuff you're emitting will be flushed in the next message.

Also, the part of the code above, would benefit to be split and have the logging part be wrapped in a `if (Log.isEnabled())`. Not urgent, but let's not add more un-guarded usages of `Log`.